### ISSUE TITLE
CLIP/SigLIP-style image encoder evolution

### ISSUE THREAD

**Issue by PatrickNercessian** [VectorialAI/evolving-dnn#33]
Wrapping self-attention to prevent routing around masking (thus breaking causality) made the whole evolution not expressive enough. We're not getting any good results.

Instead of doing AR decoding transformers, let's switch to doing image encoding. We can use SigLIP as the loss function, and our fitness with be negative validation loss.

Let's have each individual consist of two separate components that evolve (i.e. the text encoder and the image encoder)

To stay consistent with HF datasets, we can use `nlphuji/flickr30k`.
